### DISCUSSION TITLE
How to reset train data loader every epoch end??

### DISCUSSION THREAD

**Discussion by elephantoid** [Lightning-AI/pytorch-lightning#17158] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
currently, I updated PL from 1.6.5 to 2.0.0.
in past version, I use trainer.reset_train_dataloader() to reset and resampling dataloader every epoch.
why i've used that function is I want to reset only train loader in fit stage. 
so I made a custom callback and run when on_train_epoch_end
but updated version, reset_train_dataloader is deprecated.

how can i do or approach like reset_train_dataloader behavior?

**Comment by Benjamin-Barda** [comment]
There is the Trainer Flag https://lightning.ai/docs/pytorch/stable/common/trainer.html#reload-dataloaders-every-n-epochs)
